### PR TITLE
10.0 Make sync method action more user friendly

### PR DESCRIPTION
**Pull request by thinkwelltwd**
The SQL constraint gives the user no idea how
to rectify the problem.

Instead, sync all the Contacts that can be, and
collect the ones that can't into a list to
display to the user in a UserError.

### CODE REVIEW

**Comment by thinkwelltwd** [comment]
This is going to fail tests, as it won't raise a ValidationError, but rather than fix the test at this point, I thought I'd post the commit and ask if there's any reason this is not an acceptable fix.

I'll be happy to change the test to check for UserError if the overall approach is satisfactory.

File: mass_mailing_list_dynamic/models/mail_mass_mailing_list.py [hunk 1]
```diff
@@ -55,15 +58,27 @@ def action_sync(self):
...
-                Contact.create({
-                    "list_id": one.id,
-                    "partner_id": partner.id,
-                })
+                try:
+                    Contact.create({
+                        "list_id": one.id,
+                        "partner_id": partner.id,
+                    })
+                    self.env.cr.commit()
```
Comment (pedrobaeza) [inline mass_mailing_list_dynamic/models/mail_mass_mailing_list.py]: Never commit inside a transaction. Instead, you can use a savepoint:
```python
with self.env.cr.savepoint():
  # do stuff
```

**Comment by thinkwelltwd** [comment]
Sorry for this taking so long; switched to using `self.env.cr.savepoint()` as requested. Does this pass muster?

**Comment by thinkwelltwd** [comment]
Does this pass muster now?

**Pull request closed without merging**